encoding.c

```diff
@@ -89,6 +89,7 @@ static rb_encoding *global_enc_ascii,
 
 
 #define ENC_DUMMY_FLAG (1<<24)
+#define ENC_ASCIISAFE_FLAG (1<<25)
 #define ENC_INDEX_MASK (~(~0U<<24))
```

Comment:
Should `ENC_INDEX_MASK` be updated to `(~(~0U<<25))`?

Reply:
No, that's the mask for bits 0..23 which are used to store the index in `ruby_encoding_index`